parsers/powershell.c

```diff
@@ -602,6 +618,41 @@ static bool parseClass (tokenInfo *const token)
 	return readNext;
 }
 
+/* parse a enum
+ *
+ * 	enum EnumName {}
+ */
+static bool parseEnum (tokenInfo *const token)
+{
+	bool readNext = true;
+	tokenInfo *nameFree = NULL;
+
+	readToken (token);
+
+	if (token->type != TOKEN_IDENTIFIER)
+		return false;
+
+	nameFree = newToken ();
+	copyToken (nameFree, token, true);
+	readToken (token);
+
+	makeEnumTag (nameFree);
```

Comment:
Do we need `nameFree` token?

At line 635 and 636, you make a backup of `token`: `nameFree`.
Then, at line 637, read new data to `token`.
Finally, you did make a tag for `nameFree`.
I think we can do it without `nameFree`.

```
635:  makeEnumTag(token);
636:  readToken(token);
```

Reply:
(*1)

```
635: makeEnumTag(token);
636: vString *nameFree = vStringCopyNewCopy(token->string);
637: readToken(token);
```

See (*2).

Reply:
I understand your comment.
I noticed that there is an unwanted token.
I would like to delete those tokens.

I created parseEnum() based on parseClass().
It appears that parseClass() has an unwanted token as well.
I hope to change this as well in the near future.

Reply:
I understood that in this pull request, you want me to include a commit to fix parseClass() first, and then a commit to add parseEnum().

If you mean that you want me to create a new pull request, please let me know. I will do so.

Reply:
YES. I want you to create a new pull request for fixing parseClass().
I will merge it quickly. Then you can create another pull request adding parseEnum().

Reply:
Doing these things in a pull request is also acceptable as far as commits are separated.